code/Model/SiteTree.php

```diff
@@ -2612,8 +2612,10 @@ public function onAfterRevertToLive()
 
         // Need to update pages linking to this one as no longer broken
         foreach ($stageSelf->DependentPages() as $page) {
-            /** @var SiteTree $page */
-            $page->writeWithoutVersion();
+            if ($page instanceof SiteTree) {
+                /** @var SiteTree $page */
+                $page->writeWithoutVersion();
+            }
```

Comment:
```suggestion
            if ($page->hasExtension(Versioned::class)) {
                /** @var Versioned $page */
                $page->writeWithoutVersion();
            }
```
That method is provided by the `Versioned` extension, so lets allow this to work for any versioned record instead. This will avoid regressions if people were doing funky things like having `DataObject` records that behave like pages but aren't `SiteTree` explicitly.

Reply:
Updated